scripts/stac/imagery/item.py

```diff
@@ -0,0 +1,52 @@
+from typing import Any, Dict, List, Optional
+
+from scripts.stac.util import checksum
+from scripts.stac.util.stac_extensions import StacExtensions
+
+PYSTAC_VERSION = "1.0.0"
+
+
+class ImageryItem:
+    stac: Dict[str, Any]
+
+    def __init__(self, id_: Optional[str] = None, path: Optional[str] = None, stac: Optional[Dict[str, Any]] = None) -> None:
+        if stac:
+            self.stac = stac
+        elif id_ and path:
+            self.stac = {
+                "type": "Feature",
+                "stac_version": PYSTAC_VERSION,
+                "id": id_,
+                "links": [
+                    {"rel": "self", "href": f"./{id_}.json", "type": "application/json"},
+                ],
+                "assets": {
+                    "visual": {
+                        "href": path,
+                        "type": "image/tiff; application:geotiff; profile:cloud-optimized",
+                        "file:checksum": checksum.multihash_as_hex(path),
+                    }
+                },
+                "stac_extensions": [StacExtensions.file.value],
+            }
+        else:
+            raise Exception("incorrect initialising parameters must have 'stac' or 'id_ and path'")
+
+    def update_datetime(self, start_datetime: str, end_datetime: str) -> None:
+        self.stac["properties"] = {
+            "start_datetime": start_datetime,
+            "end_datetime": end_datetime,
+            "datetime": None,
+        }
+
+    def update_spatail(self, geometry: List[List[float]], bbox: List[float]) -> None:
+        self.stac["geometry"] = {"type": "Polygon", "coordinates": [geometry]}
+        self.stac["bbox"] = bbox
+
+    def add_collection(self, title: str, path: str) -> None:
```

Comment:
You could pass the `collection` object and manage getting the needed information from it in here